src/WebJobs.Extensions/Extensions/Timers/Scheduling/CronSchedule.cs

```diff
@@ -32,13 +61,133 @@ internal CrontabSchedule InnerSchedule
+            //    - Run at 01:00-7, 01:30-7, 01:00-8, 01:30-8, 02:00-8, 02:30-8, etc.
+            //
+            // For "point-in-time" schedules, we only want to run them once (i.e a 1:30 trigger only runs once on this day). Using an
+            // "every day at 01:30" schedule and Pacific time offset as an example in 2018, we'd want to:
+            //    - Run on November 3rd at 01:30-7
+            //    - Run on November 4th at 01:30-7 (and not run at 1:30-8)
+            //    - Run on November 5th at 01:30-8
+            if (!IsInterval && isNextAmbiguous && next.Offset != now.Offset)
+            {
+                // "Fall Back" Scenario 1 -- Point-in-time schedule where the next time is ambiguous and the offsets have changed.
+                //   This means that "now" is in DST and "next" is in Standard Time. Since we never want to run an ambiguous
+                //   point-in-time schedule on Standard time, move the offset back to the Daylight offset.
+                var offsetDiff = next.Offset - now.Offset;
+                adjusted = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), timeZone);
+                Logger.DstAmbiguousTime(_logger, now, _cronSchedule.ToString(), next, TimeZoneInfo.Local.DisplayName, adjusted.Value);
+            }
+            else if (!IsInterval && isNextAmbiguous && next.Offset == now.Offset)
+            {
+                // "Fall Back" Scenario 2 -- Point-in-time where "next" is ambiguous and "now" and "next" are in Standard time. This can happen
+                //   when the timer starts and we are already past the "fall back" point. For example, if the trigger starts up at
+                //   01:29-8 and we have a "every day at 01:30" schedule, we have to assume that we've already run it at 01:30-7 and
+                //   therefore calculate the time after this.
+                var nextDateTime = next.LocalDateTime;
+
+                while (timeZone.IsAmbiguousTime(nextDateTime))
+                {
+                    nextDateTime = _cronSchedule.GetNextOccurrence(nextDateTime);
+                }
+
+                adjusted = new DateTimeOffset(nextDateTime);
+                Logger.DstAmbiguousTimeSecondOccurrence(_logger, now, _cronSchedule.ToString(), next, TimeZoneInfo.Local.DisplayName, adjusted.Value);
+            }
+            else if (IsInterval && (isNowAmbiguous || isNextAmbiguous) && next.Offset != now.Offset)
+            {
+                // "Fall Back" Scenario 3 -- Interval schedule where either point is ambiguous and we're crossing the offset boundary. For example, 
+                //   an "every 30 minute" interval in Pacific time:
+                //   - 00:30-7 -> 01:00-8 (because all ambiguous "next" use Standard time) -> adjust back 01:00-7
+                //   - 01:30-7 -> 02:00-8 -> adjust back to 01:00-8
+                var offsetDiff = next.Offset - now.Offset;
+                adjusted = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), timeZone);
```

Comment:
To be sure I follow, you're saying:

next=01:00-8
now=00:30-7
offsetDiff=8-7=1
adjusted=01:00-8 + 1 = 01:00-7

That example makes sense to me. But I don't see how you get 01:00-8 from 01:30-7. Is TimeZoneInfo.ConvertTime doing this internally for you?

Reply:
`DateTimeOffset` represents an exact point in time, UTC, and the `Offset` is giving you a Local view of that time, but it knows nothing about the Time Zone or if the time is valid, for example, because none of that information exists in the way that NCronTab creates the "next".

So, during ambiguous times, the current view we have is always of the Standard/later time. In this case, with a -8 offset.

By adding the time (which in this case `(-8) - (-7) == -1` is actually subtracting an hour), we're subtracting one hour from the UTC point-in-time and telling it to give us the Offset based on the local time zone of that UTC point-in-time.

Put simply, we're an hour late and we want to move it back. But just moving it back doesn't adjust the internal offset (again, b/c it doesn't know about time zones inherently). So, we use `TimeZoneInfo.ConvertTime()` to get the offset correct. This is really just for outputting logs, though -- because I believe that just subracting an hour would work for all our interval calculations. But it makes the output look odd and it would show a time that doesn't actually exist in that time zone. (`11/4/2018 00:30-8` in this case isn't a real time; it should be `11/4/2018 0130-7`).